### PR TITLE
Add Kerbalism Configs for SpaceX Capsules, Apollo D2 GE, Apollo Block III+ SM, Soyuz MS. (Ready To Review)

### PR DESCRIPTION
**Pull request by vader111**
Add Kerbalism Configs for SpaceX Capsules, Apollo D2 GE, Apollo Block III+ SM, Soyuz MS.

### CODE REVIEW

File: GameData/KerbalismConfig/Support/ROCapsules.cfg [hunk 1]
```diff
@@ -503,6 +874,24 @@
 		%max_pressure = 0.35
 	}
 }
+@PART[ROC-D2CM]:NEEDS[ROCapsules,FeatureHabitat]:AFTER[RO-Kerbalism]
+{
+    @MODULE[Habitat]
+	{	
+		volume = 4.9     //guesstimate
+		surface = 5.3193     //m^2
```
Comment (Standecco) [inline GameData/KerbalismConfig/Support/ROCapsules.cfg:882]: Seems definitely wrong, how can a 4.9 m^3 capsule have a 5.3 m^2 surface? Look at the values that the other capsules have, a similar shape and similar volume should result in a similar surface area.

File: GameData/KerbalismConfig/Support/ROCapsules.cfg [hunk 2]
```diff
@@ -503,6 +874,24 @@
...
+		max_pressure = 0.35
+	}
+}
+@PART[ROC-D2MissionModule1]:NEEDS[ROCapsules,FeatureHabitat]:AFTER[RO-Kerbalism]
+{
+    @MODULE[Habitat]
+	{	
+		volume = 3.1        //guesstimate
+		surface = 3.728         //m^2
```
Comment (Standecco) [inline GameData/KerbalismConfig/Support/ROCapsules.cfg:891]: Again, a bit unlikely.

File: GameData/KerbalismConfig/Support/ROCapsules.cfg [hunk 3]
```diff
@@ -548,29 +937,40 @@
...
+		max_pressure = 0.35
+	}
+}
 @PART[ROC-ApolloCM]:NEEDS[ROCapsules,FeatureHabitat]:AFTER[zzzKerbalism]
 {
 	@MODULE[Habitat]
 	{	
-		%volume = 5.9
-		%surface = 15.71	 //guesstimate
+		%volume = 6.2	//m^3
```
Comment (Standecco) [inline GameData/KerbalismConfig/Support/ROCapsules.cfg:953]:
```suggestion
		%volume = 5.9	//m^3
```
Previous value was sourced, 6.2 is accurate for the proposed block III variant, but not for the actual Apollo CM.

File: GameData/KerbalismConfig/Support/ROCapsules.cfg [hunk 4]
```diff
@@ -548,29 +937,40 @@
...
+		%surface = 3.3749	 //m^2
```
Comment (Standecco) [inline GameData/KerbalismConfig/Support/ROCapsules.cfg:954]: Definitely impossible, the previous surface value was reasonably accurate.
```suggestion
		%surface = 15.71	 //m^2
```

File: GameData/KerbalismConfig/Support/ROCapsules.cfg [hunk 5]
```diff
@@ -548,29 +937,40 @@
...
 	}
 }
 @PART[ROC-ApolloCMBlockIII]:NEEDS[ROCapsules,FeatureHabitat]:AFTER[zzzKerbalism]
 {
 	@MODULE[Habitat]
 	{
-		%volume = 6.2	//override default values to account for extra space with removed cargo racks
-		%surface = 15.71	 //guesstimate
+		%volume = 6.2	//m^3
+		%surface = 3.3749	 //m^2
```
Comment (Standecco) [inline GameData/KerbalismConfig/Support/ROCapsules.cfg:963]: Same as before.

File: GameData/KerbalismConfig/Support/ROCapsules.cfg [hunk 6]
```diff
@@ -548,29 +937,40 @@
...
 	}
 }
 @PART[ROC-ApolloMissionModule]:NEEDS[ROCapsules,FeatureHabitat]:AFTER[zzzKerbalism]
 {
 	@MODULE[Habitat]
 	{
-		%volume = 16.8	//According to ETS wiki
-		%surface = 26.4	//guesstimate
+		%volume = 16.8	//m^3
+		%surface = 6.5595	//m^2
```
Comment (Standecco) [inline GameData/KerbalismConfig/Support/ROCapsules.cfg:972]: Same as before. I also wouldn't remove the previous comments, they're useful enough. Repeating m^3 and m^2 over and over is redundant

File: GameData/KerbalismConfig/Support/ROCapsules.cfg [hunk 7]
```diff
@@ -585,8 +985,9 @@
 {
 	@MODULE[Habitat]
 	{
-		%volume = 16.8	//According to ETS wiki
-		%surface = 33.0	//guesstimate
+		%volume = 20.8	//m^3 ~ Guess
```
Comment (Standecco) [inline GameData/KerbalismConfig/Support/ROCapsules.cfg:988]:
```suggestion
		%volume = 16.8	//According to ETS wiki
```
The wiki had the mission module IV as having the same habitable volume, but with extra cargo space.

File: GameData/KerbalismConfig/Support/ROCapsules.cfg [hunk 8]
```diff
@@ -585,8 +985,9 @@
...
+		%surface = 7.5633	//m^2 ~ Guess
```
Comment (Standecco) [inline GameData/KerbalismConfig/Support/ROCapsules.cfg:989]: Same as before, this surface value is impossible for that volume.

File: GameData/KerbalismConfig/Support/ROCapsules.cfg [hunk 9]
```diff
@@ -601,26 +1002,47 @@
 {
 	@MODULE[Habitat]
 	{
-		%volume = 8.95
-		%surface = 35
+		%volume = 9	//m^3
+		%surface = 4.3267	//m^2
```
Comment (Standecco) [inline GameData/KerbalismConfig/Support/ROCapsules.cfg:1006]:
```suggestion
		%surface = 35
```
Again, impossible for that internal volume.

File: GameData/KerbalismConfig/Support/ROCapsules.cfg [hunk 10]
```diff
@@ -601,26 +1002,47 @@
 {
 	@MODULE[Habitat]
 	{
-		%volume = 8.95
-		%surface = 35
+		%volume = 9	//m^3
```
Comment (Standecco) [inline GameData/KerbalismConfig/Support/ROCapsules.cfg:1005]: The old value was sourced. The difference is almost insignificant, but do you have a reason for the change?

File: GameData/KerbalismConfig/Support/ROCapsules.cfg [hunk 11]
```diff
@@ -601,26 +1002,47 @@
...
 	}
 }
 @PART[ROC-CSTCM]:NEEDS[ROCapsules,FeatureHabitat]:AFTER[zzzKerbalism]
 {
 	@MODULE[Habitat]
 	{
-		%volume = 11
-		%surface = 35	//guess, very similar dimensions to Orion
+		%volume = 11	//m^3
+		%surface = 4.9461	//m^2
```
Comment (Standecco) [inline GameData/KerbalismConfig/Support/ROCapsules.cfg:1015]:
```suggestion
		%surface = 35	//m^2
```
Impossible for that internal volume.

File: GameData/KerbalismConfig/Support/ROCapsules.cfg [hunk 12]
```diff
@@ -601,26 +1002,47 @@
 {
 	@MODULE[Habitat]
 	{
-		%volume = 8.95
-		%surface = 35
+		%volume = 9	//m^3
+		%surface = 4.3267	//m^2
+		%max_pressure = 0.35
```
Comment (Standecco) [inline GameData/KerbalismConfig/Support/ROCapsules.cfg:1007]: There are many of these, but almost all of these capsules (everything after Apollo , and everything that is Soviet in general) were used at 1 atm of pressure, so saying that the max pressure these could handle is just 0.35 atm is wrong. There are many more occurrences of this, they should all be brought in line.
```suggestion
```

File: GameData/KerbalismConfig/System/Science/HardDriveConfigs.cfg [hunk 13]
```diff
@@ -181,6 +181,12 @@ KERBALISM_HDD_SIZES
 		samples = 10
 	}
 
+	dragon_cst
+	{
+		data = 64000								//64 GB, SDD ~ Guess due to date of capsules & tech
+		samples = 3
```
Comment (Standecco) [inline GameData/KerbalismConfig/System/Science/HardDriveConfigs.cfg:187]: A bit short on samples? These capsules can return a lot of mass, definitely more than 3 samples worth.

File: GameData/KerbalismConfig/Support/ROCapsules.cfg [hunk 14]
```diff
@@ -601,26 +1002,47 @@
...
 // ============================================================================
 // SCIENCE
 // ============================================================================
 // crew experiments
 
 @PART[ROC-MercuryCM|ROC-MercuryCMBDB|ROC-VostokCapsule]:NEEDS[ROCapsules,FeatureScience,RP-0]:BEFORE[RP-0-Kerbalism] { %capsuleTier = Basic }
-@PART[ROC-GeminiCM|ROC-GeminiCMBDB|ROC-VoskhodCapsule|ROC-DynaBody]:NEEDS[ROCapsules,FeatureScience,RP-0]:BEFORE[RP-0-Kerbalism]{ %capsuleTier = SecondGen }
-@PART[ROC-ApolloCM|ROC-ApolloCMBlockIII|ROC-OrionCM]:NEEDS[ROCapsules,FeatureScience,RP-0]:BEFORE[RP-0-Kerbalism] { %capsuleTier = Mature }
+@PART[ROC-GeminiCM|ROC-GeminiCMBDB|ROC-D2MissionModule1|ROC-VoskhodCapsule|ROC-DynaBody]:NEEDS[ROCapsules,FeatureScience,RP-0]:BEFORE[RP-0-Kerbalism]{ %capsuleTier = SecondGen }
+@PART[ROC-ApolloCM|ROC-D2MissionModule2|ROC-ApolloCMBlockIII|ROC-OrionCM|ROC-Dragon2CM|ROC-Dragon2CMLunar|ROC-StarshipLunarCM]:NEEDS[ROCapsules,FeatureScience,RP-0]:BEFORE[RP-0-Kerbalism] { %capsuleTier = Mature }
```
Comment (Standecco) [inline GameData/KerbalismConfig/Support/ROCapsules.cfg:1045]: I wouldn't refrain from giving StarshipLunar a Lab module, it's basically a space station on its own

File: GameData/KerbalismConfig/Support/ROCapsules.cfg [hunk 15]
```diff
@@ -601,26 +1002,47 @@
...
+		%surface = 4.9461	//m^2
+		%max_pressure = 0.35
 	}
 }
+@PART[ROC-Dragon2CM|ROC-Dragon2CMLunar]:NEEDS[ROCapsules,FeatureHabitat]:AFTER[zzzKerbalism]
+{
+	@MODULE[Habitat]
+	{
+		%volume = 11.8064	// m^3 ~ Calculated from the Internal Model Volume
+		%surface = 5.185	// m^2 ~ Calculated from the Internal Model Volume
```
Comment (Standecco) [inline GameData/KerbalismConfig/Support/ROCapsules.cfg:1024]: Not sure what you mean by "internal model volume", but not even a sphere can have that low of a surface area with that high of a volume, same as all the other previous values.

File: GameData/KerbalismConfig/Support/ROCapsules.cfg [hunk 16]
```diff
@@ -601,26 +1002,47 @@
...
+		%max_pressure = 0.35
```
Comment (Standecco) [inline GameData/KerbalismConfig/Support/ROCapsules.cfg:1025]: Again, these can hold 1 atm of pressure just fine.
```suggestion
```

File: GameData/KerbalismConfig/Support/ROCapsules.cfg [hunk 17]
```diff
@@ -601,26 +1002,47 @@
...
+	}
+}
+@PART[ROC-StarshipLunarCM]:NEEDS[ROCapsules,FeatureHabitat]:AFTER[zzzKerbalism]
+{
+	@MODULE[Habitat]
+	{
+		%volume = 297.531	// m^3 ~ Calculated from the Internal Model Volume
+		%surface = 44.5678	// m^2 ~ Calculated from the Internal Model Volume
+		%max_pressure = 0.35
```
Comment (Standecco) [inline GameData/KerbalismConfig/Support/ROCapsules.cfg:1034]: Surface area looks sane, but max_pressure should go.
```suggestion
```

**Review by Standecco — CHANGES_REQUESTED** [review]

**Comment by vader111** [comment]
(Ready To Review)

**Comment by vader111** [comment]
Split into https://github.com/Standecco/ROKerbalism/pull/92 https://github.com/Standecco/ROKerbalism/pull/93 https://github.com/Standecco/ROKerbalism/pull/94 https://github.com/Standecco/ROKerbalism/pull/95 https://github.com/Standecco/ROKerbalism/pull/96

**Pull request closed without merging**